test/TensorFlow/debug_scope.swift

```diff
@@ -6,5 +6,6 @@
 
 import TensorFlow
 func test() {
-  _ = Tensor(1)
+  let x: Tensor<Float> = [1, 2, 3, 4] // expected-error {{array input is not a constant array of tensors}}
```

Comment:
will these errors go away once we finish deabstraction, move up the partition pass and turn on -Onone?

I'm not sure whether this test here is still relevant to debug scope -- do we want to rename it to something like "deabstraction_finished"?

I guess a test that's relevant to debug scope would be to check if an error message is produced at the right source code location.

Reply:
Agreed, I'm not sure what this test change is about?

Reply:
`test/TensorFlow/debug_scope.swift` was added solely to test that deabstraction passes debug scope verification with `-Onone`. The test is made more extensive in this PR because the [previous debug scope workaround](https://github.com/apple/swift/pull/17797) was sufficient for handling the old version of the test but not the new version.

As soon as there's another test that runs deabstraction with `-Onone`, `debug_scope.swift` can be deleted. `deabstraction_finished.swift` almost fits the bill but currently crashes when `-O` is removed.